app/v2/retrievePeriodsOfAccount/model/response/RetrievePeriodsOfAccountResponse.scala

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.retrievePeriodsOfAccount.model.response
+
+import play.api.libs.json.{Json, OWrites, Reads}
+import v2.common.models.PeriodsOfAccountDates
+
+case class RetrievePeriodsOfAccountResponse(periodsOfAccount: Option[Boolean], periodsOfAccountDates: Option[Seq[PeriodsOfAccountDates]])
+
+object RetrievePeriodsOfAccountResponse {
+
+  implicit val reads: Reads[RetrievePeriodsOfAccountResponse] = Json.reads[RetrievePeriodsOfAccountResponse]
+
+  implicit val writes: OWrites[RetrievePeriodsOfAccountResponse] = (o: RetrievePeriodsOfAccountResponse) =>
+    (o.periodsOfAccount, o.periodsOfAccountDates) match {
+      case (None, Some(dates)) => Json.obj("periodsOfAccount" -> true, "periodsOfAccountDates" -> dates)
+      case _                   => Json.obj("periodsOfAccount" -> false)
```

Comment:
is this missing cases. Looks like it will just return Json.obj("periodsOfAccount" -> false) if periodsOfAccount is returned and is true. the case (Some(true), Some(dates)) can some tests for this scenario be written as well.